internal/dao/stubs/context_getters.go

```diff
@@ -0,0 +1,43 @@
+package stubs
+
+import (
+	"context"
+	"errors"
+
+	"github.com/RHEnVision/provisioning-backend/internal/models"
+)
+
+type daoStubCtxKeyType int
+
+const (
+	accountCtxKey daoStubCtxKeyType = iota
+	pubkeyCtxKey  daoStubCtxKeyType = iota
+)
+
+var contextReadError = errors.New("the message")
+
+func WithPubkeyDao(parent context.Context, init_store []*models.Pubkey) context.Context {
+	ctx := context.WithValue(parent, pubkeyCtxKey, &PubkeyDaoStub{init_store})
+	return ctx
+}
+
+func getPubkeyDaoStub(ctx context.Context) (pkdao *PubkeyDaoStub, err error) {
+	var ok bool
+	if pkdao, ok = ctx.Value(pubkeyCtxKey).(*PubkeyDaoStub); !ok {
+		err = contextReadError
+	}
+	return
+}
+
+func WithAccountDaoOne(parent context.Context) context.Context {
+	ctx := context.WithValue(parent, accountCtxKey, &AccountDaoStub{})
+	return ctx
+}
+
+func getAccountDaoStub(ctx context.Context) (accdao *AccountDaoStub, err error) {
+	var ok bool
+	if accdao, ok = ctx.Value(accountCtxKey).(*AccountDaoStub); !ok {
+		err = contextReadError
+	}
+	return
```

Comment:
Uh, oh, well, okay. Cannot say I love it! But it's fine.

For the record, this is a "hidden feature" of Go: https://go.dev/tour/basics/7

Reply:
Got rid of it :)